src/test/java/uk/nhs/hee/tis/trainee/credentials/service/IssuanceServiceTest.java

```diff
@@ -137,4 +191,98 @@ void shouldReturnRedirectUriWhenGatewayRequestSuccessful() {
 
     assertThat("Unexpected URI.", uri, is(Optional.of(redirectUri)));
   }
+
+  @Test
+  void shouldNotSaveToRepositoryWhenCredentialNotIssued() {
+    issuanceService.completeCredentialVerification(null, STATE_VALUE, "error", "error_description");
+    verifyNoInteractions(credentialMetadataRepository);
+  }
+
+  @ParameterizedTest
+  @MethodSource("credentialMethodSource")
+  void shouldSaveToRepositoryWhenCredentialIssued(CredentialDto credentialData) {
+    UUID nonce = UUID.randomUUID();
+    Claims claimsIssued = new DefaultClaims();
+    claimsIssued.put("nonce", nonce.toString());
+    claimsIssued.put("SerialNumber", CREDENTIAL_ID);
+    claimsIssued.put("iat", ISSUED_AT.getEpochSecond());
+    claimsIssued.put("exp", EXPIRES_AT.getEpochSecond());
+
+    // TODO: this is quite brittle
```

Comment:
Are we ok to leave this for now, or would be worth refactoring this test?

Reply:
I wasn't sure what in particular you considered brittle, something to go over. I'll merge this to your branch and let sonarcloud flag it again.

Reply:
I think it just felt like the test needed to know quite a lot about how the method under test worked internally, but happy to leave it until sonarcloud squeeks :)